config/v1/types_dns.go

```diff
@@ -96,3 +103,31 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
```

Comment:
Worth adding `+kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'AWS' ?  has(self.aws) : !has(self.aws)",message="aws configuration is required when platform is AWS, and forbidden otherwise"` validation as in https://github.com/openshift/api/blob/master/machine/v1/types_controlplanemachineset.go#L230-L232. Even thought there's only `AWS` field for now, it helps to remind the validation is needed for new platforms as well.

Reply:
Also worth introducing integration tests. E.g. https://github.com/openshift/api/blob/master/machine/v1/stable.controlplanemachineset.aws.testsuite.yaml

Reply:
> aws configuration is required when platform is AWS, and forbidden otherwise

Question about this: my intent was not to make aws configuration required on aws. Is that preferred? If we did that, how does the DNS config get updated when upgrading AWS clusters?

My intent was that the AWS platform would be an optional field, so DNS config in existing AWS clusters would not need to be updated.

Reply:
David answered my question. On upgrade, platform type  for aws clusters will be "". This logic makes sense to me now.